### ISSUE TITLE
[awscli/1.11.4] `aws s3 sync` failed to delete any local files

### ISSUE THREAD

**Issue by yoheimuta** [aws/aws-cli#2233]
I think this issue is a bug or a breaking compatibility change (not describing anywhere), so please response it 🙇 

I read https://github.com/aws/aws-cli/blob/develop/CONTRIBUTING.rst#reporting-issues.

> A description of the problem/suggestion.

Our aws s3 sync command could not work well as before, after upgrading awscli from 1.11.3(any previous ones) to 1.11.4.

> The specific AWS CLI commands you are running. Please include debug logs for these commands by appending the --debug option to each command. Be sure to remove any sensitive information from the debug logs.

```
# Contents between the bucket and the local are different.
$ aws s3 ls s3://mtburn-test # empty
$ ll /tmp/test
total 0
0 -rw-r--r--  1 great  wheel     0B 10 14 17:00 will_be_delete.txt

# Expected to delete a local file, but failed
$ aws s3 sync \
s3://mtburn-test \
/tmp/test \
--delete; echo $?
delete failed: s3:///tmp/test/will_be_delete.txt Parameter validation failed:
Invalid bucket name "": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$"
1
```

> The AWS CLI version you are using aws --version.

```
$ aws --version
aws-cli/1.11.4 Python/2.7.9 Darwin/15.6.0 botocore/1.4.62
```

- To be sure, I could reproduce this issue in Ubuntu 14.04 and MacOSX 10.11.6.

> The first thing an AWS CLI developer will do is try to reproduce the issue you are seeing, so try to reduce your issue to the smallest possible set of steps that demonstrate the issue. This will lead to quicker resolution of your issue.

```
# Create a test bucket
$ aws s3 mb s3://mtburn-test
make_bucket: mtburn-test

# Create a local file
$ touch /tmp/test/will_be_delete.txt
$ ll /tmp/test/
total 0
0 -rw-r--r--  1 great  wheel     0B 10 14 17:00 will_be_delete.txt

# Install v1.11.3
$ sudo pip install awscli==1.11.3
...
$ aws s3 sync \
s3://mtburn-test \
/tmp/test \
--delete; echo $?
delete: ../../../tmp/test/will_be_delete.txt
0

# Install v1.11.4
$ sudo pip install awscli==1.11.4
...
$ touch /tmp/test/will_be_delete.txt
$ aws s3 sync \
s3://mtburn-test \
/tmp/test \
--delete; echo $?
delete failed: s3:///tmp/test/will_be_delete.txt Parameter validation failed:
Invalid bucket name "": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$"
1
$ ll /tmp/test/
total 0
0 -rw-r--r--  1 great  wheel     0B 10 14 17:00 will_be_delete.txt
```

**Comment by josephtate** [comment]
It's simpler than this:

Have a bucket with some contents and sync it locally:

```
$ aws s3 sync --delete s3://bucket/path/ somelocalpath/
... completes ..
$ touch somelocalpath/foo.txt
$ aws s3 sync --delete s3://bucket/path/ somelocalpath/
... Parameter validation failed error as above ...
```

**Comment by kyleknap** [comment]
I can confirm the issue. For now, I would recommend downgrading to 1.11.3 while we look into fixing this.

**Comment by kyleknap** [comment]
This should now be fixed in 1.11.5. Thank you for reporting the issue.

**Comment by yoheimuta** [comment]
Thank you!
